frontend/ui/widget/titlebar.lua

```diff
@@ -0,0 +1,250 @@
+local Device = require("device")
+local Font = require("ui/font")
+local Geom = require("ui/geometry")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local HorizontalSpan = require("ui/widget/horizontalspan")
+local IconButton = require("ui/widget/iconbutton")
+local LineWidget = require("ui/widget/linewidget")
+local OverlapGroup = require("ui/widget/overlapgroup")
+local Size = require("ui/size")
+local TextWidget = require("ui/widget/textwidget")
+local UIManager = require("ui/uimanager")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local VerticalSpan = require("ui/widget/verticalspan")
+local Screen = Device.screen
+
+local TitleBar = OverlapGroup:extend{
+    width = nil, -- default to screen width
+    fullscreen = false, -- larger font and small adjustments if fullscreen
+    align = "center",
+    with_bottom_line = false,
+
+    title = "",
+    subtitle = nil,
+    title_face = nil, -- if not provided, one of these will be used:
+    title_face_fullscreen = Font:getFace("tfont"),
```

Comment:
We now have
https://github.com/koreader/koreader/blob/eae07af1574013ead4bb44a7fe1d835f97e108a7/frontend/apps/filemanager/filemanager.lua#L162
https://github.com/koreader/koreader/blob/eae07af1574013ead4bb44a7fe1d835f97e108a7/frontend/ui/widget/menu.lua#L672

Reply:
Oh, sh*t...
You added `Screen:scaleBySize(3)`. With `smalltfont`, my ugly:
https://github.com/koreader/koreader/blob/4c603cf7e820a023a6c961d639b772413da9a9ef/frontend/ui/widget/titlebar.lua#L88-L96
is wrong, and would be even more ugly if I add `Screen:scaleBySize(3)` in it...

But this `Screen:scaleBySize(3)` relates to https://github.com/koreader/koreader/pull/8603#issuecomment-1003901741. Any thought on it ?

May be we should just not use any fixed top padding, and try to compute it from the TextWidget baseline, vs. icons top padding + some imaginary baseline for the icons, something like 80% of icon height.

Reply:
I see on your screenshots that the right cross is not diagonally directed to the corner.
Maybe setting `title_face_fullscreen = Font:getFace("smalltfont")` would be better?

Reply:
No, that's because of my doing a screenshot and croping the area by hands, to not take the borders of my window with it :)
Your trick of using the same padding on top & left/right edge is enough to ensure it's diagonally aligned, I think.

Using smalltfont, the title feels above the icons - or the icons feel too low:
![image](https://user-images.githubusercontent.com/24273478/148067817-66f950af-baad-40c6-a87d-506365b789c9.png)

Top one is my (with Page browser renamed to Bookmarks, just for comparisons) which feels wrongly aligned.
Bttom one is yours, which feels correctly aligned.
So, I need the same padding as you - but I don't want to hardcode it, I want something that work with any font callers may provide to TitleBar, and at least with title_face_not_fullscreen and with_bottom_line - the trick above worked quite well in a DictQuickLookup.

Reply:
Something like:
```lua
    if not self.title_top_padding then
        -- Compute it so baselines of the text and of the icons align
        -- Our icons baselines looks like they could be at 0.83 to 0.9 of their height.
        local face_height, face_baseline = self.title_face.ftface:getHeightAndAscender()
        local icon_height = math.max(left_icon_size, right_icon_size)
        local icon_baseline = icon_height * 0.80 + self.button_padding
        self.title_top_padding = Math.round(math.max(0,  icon_baseline - face_baseline))
        require("logger").warn(face_height, face_baseline, "+", self.button_padding, "vs", icon_height, icon_baseline, "=", self.title_top_padding)
    end
```
I'm hesitating on the factor, 0.75, 0.85...
I'd need to see how it looks in a title with bottom bar, but I unfortunately lost my DictQuickLookup and Calendar from yesterday :(
```
git diff frontend/... > ~/titlebar_tweaks.diff
Ctrl-R to just change diff to checkout:
git checkout frontend/... > ~/titlebar_tweaks.diff
```
and I lost the diff and my work we :(